### PR TITLE
Remove associated events and notifications when membership destroyed

### PR DESCRIPTION
**Pull request by HSalmon**
Fix for: https://github.com/loomio/loomio/issues/3675
Angular record store PR: https://github.com/loomio/angular_record_store/pull/14

### CODE REVIEW

**Comment by HSalmon** [comment]
@robguthrie, @gdpelican, could someone take a look at this (Rob I know you were involved, but it would be good to get sign-off)

**Comment by robguthrie** [comment]
Looks good but we'll need to tag and push a new version of record store.

We should document that within the record store repo when we do it.

**Comment by HSalmon** [comment]
@robguthrie, can we pair on making this happen?

**Comment by HSalmon** [comment]
@robguthrie, added test with last commit :)

**Comment by HSalmon** [comment]
...it looks like something's breaking the Travis build though

File: angular/package.json [hunk 1]
```diff
@@ -40,7 +40,7 @@
     "angular-translate": "~2.7.2",
     "angular-translate-loader-url": "~2.7.2",
     "angular-ui-bootstrap": "~> 0.14.3",
-    "angular_record_store": "~2.6.0",
+    "angular_record_store": "2.6.1",
```
Comment (gdpelican) [inline angular/package.json]: I think you can leave this at `~2.6.0`, which allows patch level changes (so, 'give me the latest up to but not including 2.7.0')

Then we won't have to change it for 2.6.2, etc.

**Comment by gdpelican** [comment]
Looks like the package is getting pulled in correctly, but would be good to see tests passing on this once Rob gets the test suite humming again, since it's a slightly bigger piece of work.

File: angular/core/models/event_model.coffee [hunk 2]
```diff
@@ -25,3 +26,7 @@ angular.module('loomioApp').factory 'EventModel', (BaseModel) ->
 
     model: ->
       @recordStore[@constructor.eventTypeMap[@eventable.type]].find(@eventable.id)
+
+    beforeRemove: ->
+      _.each @notifications(), (notification) ->
```
Comment (gdpelican) [inline angular/core/models/event_model.coffee]: I believe you can do `_.apply(@notifications(), 'remove')` here.

File: angular/core/models/membership_model.coffee [hunk 3]
```diff
@@ -36,3 +36,10 @@ angular.module('loomioApp').factory 'MembershipModel', (BaseModel, AppConfig) ->
 
     isMuted: ->
       @volume == 'mute'
+
+    events: ->
```
Comment (gdpelican) [inline angular/core/models/membership_model.coffee]: Do we need this?

File: angular/core/models/membership_model.coffee [hunk 4]
```diff
@@ -36,3 +36,10 @@ angular.module('loomioApp').factory 'MembershipModel', (BaseModel, AppConfig) ->
...
+      @recordStore.events.find()
+
+    beforeRemove: ->
+      _.each @recordStore.events.find('eventable.type': 'membership', 'eventable.id': @id), (e) ->
```
Comment (gdpelican) [inline angular/core/models/membership_model.coffee]: `_.apply` should work here as well.

Comment (gdpelican) [inline angular/core/models/membership_model.coffee]: Er, sorry. `_.invoke` it's called.

**Comment by HSalmon** [comment]
@gdpelican, I've implemented your feedback - thanks. @robguthrie, is this ready to go d'you reckon?

**Comment by HSalmon** [comment]
Can I get a green apple?

**Comment by gdpelican** [comment]
I'm happy to 🍏  it. Do you feel on top of the angular record store changes? If so merge away.

**Pull request merged**